pkg/trace/stats/aggregation.go

```diff
@@ -21,107 +22,40 @@ type Aggregation struct {
 	Env        string
 	Resource   string
 	Service    string
+	Type       string
 	Hostname   string
-	StatusCode string
+	StatusCode uint32
 	Version    string
 	Synthetics bool
 }
 
+func getStatusCode(s *pb.Span) uint32 {
+	if s.Meta[tagStatusCode] == "" {
+		return 0
+	}
+	c, err := strconv.Atoi(s.Meta[tagStatusCode])
+	if err != nil {
+		log.Errorf("Invalid status code %s. Using 0.", c)
+		return 0
+	}
+	return uint32(c)
+}
+
 // NewAggregationFromSpan creates a new aggregation from the provided span and env
-func NewAggregationFromSpan(s *pb.Span, env string) Aggregation {
+func NewAggregationFromSpan(s *pb.Span, env string, defaultHostname string) Aggregation {
 	synthetics := strings.HasPrefix(s.Meta[tagOrigin], "synthetics")
-
+	hostname := s.Meta[tagHostname]
+	if hostname == "" {
```

Comment:
Maybe it can never happen, but it crossed my mind so leaving a comment. Let's use `GetMeta` here to avoid any surprises :)
```suggestion
	hostname := defaultHostname
	if v, ok := traceutil.GetMeta(s, tagHostname); ok {
		hostname = v
	}
```

Reply:
Sounds good. I added a method: `GetMetaDefault`